src/Microsoft.Data.Sqlite/SqliteDataReader.cs

```diff
@@ -297,6 +301,7 @@ public override bool IsDBNull(int ordinal)
 
         public override double GetDouble(int ordinal)
         {
+            AssertSequentialAccess(ordinal);
```

Comment:
Is it right to assume that the only effect of using `SequentialAccess` is that we force the _`GetXyzz()`_ methods to throw if you read out of order? 

The value of `SequentialAccess` is that it can help remove unnecessary overhead in certain provider implementations. I would recommend just ignoring the behavior (as I think @bricelam originally suggested) rather than adding unnecessary overhead :smile: